onnx/onnx.in.proto

```diff
@@ -196,6 +204,83 @@ message NodeProto {
   optional string doc_string = 6;
 }
 
+// Training information
+// TrainingInfoProto defines a computation graph similar to the inference one
+// (i.e., ModelProto.graph) but represents a training algorithm. Each execution
+// of a TrainingInfoProto leads to the update of some initializers in its
+// TrainingInfoProto.algorithm.initializer and ModelProto.graph.initializer.
+//
+// If there are multiple TrainingInfoProto's, sequentially executing all of
+// them once constitutes a full training algorithm step. The initializer update
+// happens at the end of each TrainingInfoProto's execution. If the targeted
+// training algorithm contains consecutive update stages (such as block
+// coordinate descent methods), the user needs to create a TrainingInfoProto
+// for each stage.
+//
+// The execution of TrainingInfoProto has two modes. The user can choose to
+// execute either "algorithm" or "initialization". The computation specified in
+// "training" improves initializers based on the given training data. The
+// computation specified in "initialization" resets the model to its
+// initial state (as if no training data has been consumed).
+//
+// Sveral inference components are usable in TrainingInfoProto; the list
+// includes
+//   1. ModelProto.graph is a callble inside TrainingInfoProto.algorithm.
+//      Each call is a NodeProto in TrainingInfoProto.algorithm.node with
+//      NodeProto.op_type == "Graph" and
+//      NodeProto.name == ModelProto.graph.name.
+//   2. Initializers in ModelProto.graph.initializer can be referenced by
+//      nodes in TrainingInfoProto.algorithm.nodes and
+//      TrainingInfoProto.initialization.nodes. It implies that training
+//      nodes cannot produce values with names identical to inference
+//      initializers.
+message TrainingInfoProto {
+  // This field is a graph that describes a way to compute the initial tensors
+  // upon starting the training process. Initialization graph has no input
+  // and can have multiple outputs. Assigning those outputs to the associated
```

Comment:
Would be nice to describe what typically goes into initialization. Is it just like algorithm without inputs? I really like the comment below for algorithm, "In general, this graph contains loss node, gradient node, optimizer node, increment of iteration count, and calls to the inference graph."

Reply:
I will add
```
+  // Usually,
+  // trainable tensors in neural networks are randomly initialized.
+  // To achieve that, for each tensor, the user can put a random operator such
+  // as RandomNormal or RandomUniform in TrainingInfoProto.initialization.node
+  // and assign its random output to the specific tensor using "update_binding".
```